### DISCUSSION TITLE
Fail to start the LSP in VS Code

### DISCUSSION THREAD

**Discussion by maxno-kivra** [erlang-ls/erlang_ls#1115] [Support Requests]
Upon loading the extension I get:

```
[Error - 1:22:25 PM] Starting client failed
Launching server using command escript failed.
```

I tried to crank up the logging, i.e. set log level to debug and verbose tracing. But to no avail. It doesn't seem to log anything at all (I deleted all logs and got no new ones.).

I usually run on latest master, and tried to switch to the stock LSP. Same results. I then tried to manually call the `erlang_ls`/`els_dap` command, which both works just fine.

I'm at a lose here, how do I get it to log more? What could cause it to fail like this?

**Comment by robertoaloi** [comment]
Hi @maxno-kivra !

If I have to guess, it could be that you are using an incompatible combination of OTP / rebar3 / the escript itself. Can you check which versions are you using for all of them? Keep in mind that your editor may have a different notion of the `PATH`, so versions used from the terminal may not be the same as used by VS Code. The pre-packaged escript is compiled with OTP 21 and should work with all versions of OTP from 21 to 24. Which operating system are you using?

**Reply by maxno-kivra** [reply]
Mac OS X Big Sur on x86, my local copy is compiled with brew Erlang 24 and rebar3 3.17.0. The project I'm in is using Erlang 24 as well, but an slightly older rebar3 (3.16.1).

I tried using the stock/shipped version which didn't work, but running the LSP/DAP server by hand in the repo did work.

For some reason today it started working again. So weird. It would be nice if you could get a better error message when it fails somehow.

*Edit:* You can't close discussions it seems. Or maybe only maintainers? If so you can close this.

**Reply by maxno-kivra** [reply]
VS Code actually runs your default shell and records the PATH, so I've tried to make sure it picks up asdf. In any case, the stock/shipped LSP server should work with all the Erlangs I have.

**Reply by robertoaloi** [reply]
>  It would be nice if you could get a better error message when it fails somehow.

We are pushing upstream exactly for this. See: https://github.com/microsoft/vscode-languageserver-node/issues/824

**Reply by robertoaloi** [reply]
@maxno-kivra The latest version of the VS Code extension (still unpublished, you need to build the VSIX yourself) will show errors whenever the server cannot start correctly (https://github.com/erlang-ls/vscode/pull/114). Feel free to give it a try.
